drivers/serial/uart_bm.c

```diff
@@ -0,0 +1,1661 @@
+					   uint32_t stopbits)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	LL_USART_SetStopBitsLength(config->usart, stopbits);
+}
+
+static inline uint32_t uart_bm_get_stopbits(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	return LL_USART_GetStopBitsLength(config->usart);
+}
+
+static inline void uart_bm_set_databits(const struct device *dev,
+					   uint32_t databits)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	LL_USART_SetDataWidth(config->usart, databits);
+}
+
+static inline uint32_t uart_bm_get_databits(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	return LL_USART_GetDataWidth(config->usart);
+}
+
+static inline void uart_bm_set_hwctrl(const struct device *dev,
+					 uint32_t hwctrl)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	LL_USART_SetHWFlowCtrl(config->usart, hwctrl);
+}
+
+static inline uint32_t uart_bm_get_hwctrl(const struct device *dev)
+{
+	const struct uart_bm_config *config = dev->config;
+
+	return LL_USART_GetHWFlowCtrl(config->usart);
+}
+
+static inline uint32_t uart_bm_cfg2ll_parity(enum uart_config_parity parity)
+{
+	switch (parity) {
+	case UART_CFG_PARITY_ODD:
+		return LL_USART_PARITY_ODD;
+	case UART_CFG_PARITY_EVEN:
+		return LL_USART_PARITY_EVEN;
+	case UART_CFG_PARITY_NONE:
+	default:
+		return LL_USART_PARITY_NONE;
+	}
+}
+
+static inline enum uart_config_parity uart_bm_ll2cfg_parity(uint32_t parity)
+{
+	switch (parity) {
+	case LL_USART_PARITY_ODD:
+		return UART_CFG_PARITY_ODD;
+	case LL_USART_PARITY_EVEN:
+		return UART_CFG_PARITY_EVEN;
+	case LL_USART_PARITY_NONE:
+	default:
+		return UART_CFG_PARITY_NONE;
+	}
+}
+
+static inline uint32_t uart_bm_cfg2ll_stopbits(enum uart_config_stop_bits sb)
+{
+	switch (sb) {
+/* Some MCU's don't support 0.5 stop bits */
+#ifdef LL_USART_STOPBITS_0_5
+	case UART_CFG_STOP_BITS_0_5:
+		return LL_USART_STOPBITS_0_5;
+#endif	/* LL_USART_STOPBITS_0_5 */
+	case UART_CFG_STOP_BITS_1:
+		return LL_USART_STOPBITS_1;
+/* Some MCU's don't support 1.5 stop bits */
+#ifdef LL_USART_STOPBITS_1_5
+	case UART_CFG_STOP_BITS_1_5:
+		return LL_USART_STOPBITS_1_5;
+#endif	/* LL_USART_STOPBITS_1_5 */
+	case UART_CFG_STOP_BITS_2:
+	default:
+		return LL_USART_STOPBITS_2;
+	}
+}
+
+static inline enum uart_config_stop_bits uart_bm_ll2cfg_stopbits(uint32_t sb)
+{
+	switch (sb) {
+/* Some MCU's don't support 0.5 stop bits */
+#ifdef LL_USART_STOPBITS_0_5
+	case LL_USART_STOPBITS_0_5:
+		return UART_CFG_STOP_BITS_0_5;
+#endif	/* LL_USART_STOPBITS_0_5 */
+	case LL_USART_STOPBITS_1:
+		return UART_CFG_STOP_BITS_1;
+/* Some MCU's don't support 1.5 stop bits */
+#ifdef LL_USART_STOPBITS_1_5
+	case LL_USART_STOPBITS_1_5:
+		return UART_CFG_STOP_BITS_1_5;
+#endif	/* LL_USART_STOPBITS_1_5 */
+	case LL_USART_STOPBITS_2:
+	default:
+		return UART_CFG_STOP_BITS_2;
+	}
+}
+
+static inline uint32_t uart_bm_cfg2ll_databits(enum uart_config_data_bits db,
```

Comment:
Potentially removable API - default to 8 data bit width. I don't see us ever changing this while using a UART as the communication peripheral.